blockchain/reactor.go

```diff
@@ -122,6 +122,28 @@ func (bcR *BlockchainReactor) RemovePeer(peer *p2p.Peer, reason interface{}) {
 	bcR.pool.RemovePeer(peer.Key)
 }
 
+// respondToPeer loads a block and sends it to the requesting peer, if we have it.
+// According to the Tendermint spec, if all nodes are honest, no node should
+// be requesting for a block that's non-existent.
+func (bcR *BlockchainReactor) respondToPeer(msg *bcBlockRequestMessage, src *p2p.Peer) (queued bool) {
+	// Got a request for a block. Respond with block if we have it.
+	block := bcR.store.LoadBlock(msg.Height)
+	if block != nil {
+		msg := &bcBlockResponseMessage{Block: block}
+		return src.TrySend(BlockchainChannel, struct{ BlockchainMessage }{msg})
+	}
+
+	// Otherwise peer is asking for things we don't have.
+	// According to the Tendermint spec, if the peer asks for
+	// something that we don't have have, some of the nodes aren't honest
+	bcR.Logger.Debug("Respond", "src", src, "flagging", "potentially malicious peer",
```

Comment:
Let's simplify this message. Should be `Info` level and msg can just be `Peer asking for block we don't have`. Tags can just be `src` and `height`

Reply:
Roger that, thanks. I'll update that.